darshan-util/pydarshan/darshan/experimental/plots/plot_access_histogram.py

```diff
@@ -33,44 +33,64 @@ def plot_access_histogram(report, mod, ax=None):
     else:
         fig = None
 
-    # TODO: change to report.summary
-    if 'mod_agg_iohist' in dir(report):
-        report.mod_agg_iohist(mod)
-    else:
-        print("Cannot create summary, mod_agg_iohist aggregator is not registered with the report class.")
 
     # defaults
     labels = ['0-100', '101-1K', '1K-10K', '10K-100K', '100K-1M', '1M-4M', '4M-10M', '10M-100M', '100M-1G', '1G+']
 
-    agg = report.summary['agg_iohist'][mod]
-    # TODO: can simplify the read/write vals below after
-    # support for python 3.6 is dropped
-    read_vals = [
-        agg['READ_0_100'],
-        agg['READ_100_1K'],
-        agg['READ_1K_10K'],
-        agg['READ_10K_100K'],
-        agg['READ_100K_1M'],
-        agg['READ_1M_4M'],
-        agg['READ_4M_10M'],
-        agg['READ_10M_100M'],
-        agg['READ_100M_1G'],
-        agg['READ_1G_PLUS']
-    ]
-
-    write_vals = [
-        agg['WRITE_0_100'],
-        agg['WRITE_100_1K'],
-        agg['WRITE_1K_10K'],
-        agg['WRITE_10K_100K'],
-        agg['WRITE_100K_1M'],
-        agg['WRITE_1M_4M'],
-        agg['WRITE_4M_10M'],
-        agg['WRITE_10M_100M'],
-        agg['WRITE_100M_1G'],
-        agg['WRITE_1G_PLUS']
-    ]
-
+    agg=record['counters']
+    if mod == 'POSIX':
+        read_vals = [
+            agg['POSIX_SIZE_READ_0_100'][0],
```

Comment:
Notice how similar the counter names are here for POSIX and MPI-IO modules. They only differ in the prefix module name that is provided as input to this routine. I would get rid of this module-specific if-else logic and use string manipulation to get the desired module-specific counter names instead. Will be visually much cleaner since it's half as much code.

Reply:
yes, it will be cleaner.